File: CS/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using DevExpress.XtraEditors;

namespace WindowsApplication596
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class Form1 : System.Windows.Forms.Form
	{
		private DevExpress.XtraEditors.ListBoxControl listBoxControl1;
		private DevExpress.XtraEditors.ListBoxControl listBoxControl2;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public Form1()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.listBoxControl1 = new DevExpress.XtraEditors.ListBoxControl();
            this.listBoxControl2 = new DevExpress.XtraEditors.ListBoxControl();
            ((System.ComponentModel.ISupportInitialize)(this.listBoxControl1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.listBoxControl2)).BeginInit();
            this.SuspendLayout();
            // 
            // listBoxControl1
            // 
            this.listBoxControl1.ItemHeight = 15;
            this.listBoxControl1.Location = new System.Drawing.Point(32, 16);
            this.listBoxControl1.Name = "listBoxControl1";
            this.listBoxControl1.Size = new System.Drawing.Size(248, 288);
            this.listBoxControl1.TabIndex = 0;
            this.listBoxControl1.MouseMove += new System.Windows.Forms.MouseEventHandler(this.listBoxControl1_MouseMove);
            this.listBoxControl1.MouseDown += new System.Windows.Forms.MouseEventHandler(this.listBoxControl1_MouseDown);
            // 
            // listBoxControl2
            // 
            this.listBoxControl2.AllowDrop = true;
            this.listBoxControl2.ItemHeight = 15;
            this.listBoxControl2.Location = new System.Drawing.Point(360, 16);
            this.listBoxControl2.Name = "listBoxControl2";
            this.listBoxControl2.Size = new System.Drawing.Size(248, 288);
            this.listBoxControl2.TabIndex = 0;
            this.listBoxControl2.DragOver += new System.Windows.Forms.DragEventHandler(this.listBoxControl2_DragOver);
            this.listBoxControl2.DragDrop += new System.Windows.Forms.DragEventHandler(this.listBoxControl2_DragDrop);
            // 
            // Form1
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(768, 510);
            this.Controls.Add(this.listBoxControl1);
            this.Controls.Add(this.listBoxControl2);
            this.Name = "Form1";
            this.Text = "How to drag/drop ListBoxControl Items between ListBoxControls";
            this.Load += new System.EventHandler(this.Form1_Load);
            ((System.ComponentModel.ISupportInitialize)(this.listBoxControl1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.listBoxControl2)).EndInit();
            this.ResumeLayout(false);

		}
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

		private void Form1_Load(object sender, System.EventArgs e) {
			for (int i = 0; i < 10; i++) {
				if (i % 2 == 0)
					listBoxControl1.Items.Add("Item " + i.ToString());
				else
					listBoxControl2.Items.Add("Item " + i.ToString());				
			}
		}

		Point p = Point.Empty;

		private void listBoxControl1_MouseDown(object sender, System.Windows.Forms.MouseEventArgs e) {
			ListBoxControl c = sender as ListBoxControl;
			p = new Point(e.X, e.Y);
			int selectedIndex = c.IndexFromPoint(p);
			if (selectedIndex == -1)
				p = Point.Empty;
		}

		private void listBoxControl1_MouseMove(object sender, System.Windows.Forms.MouseEventArgs e) {
			if (e.Button == MouseButtons.Left)
				if ((p != Point.Empty) && ((Math.Abs(e.X - p.X) > 5) || (Math.Abs(e.Y - p.Y) > 5)))
					listBoxControl1.DoDragDrop(sender, DragDropEffects.Move);
		}

		private void listBoxControl2_DragOver(object sender, System.Windows.Forms.DragEventArgs e) {
			e.Effect = DragDropEffects.Move;
		}

		private void listBoxControl2_DragDrop(object sender, System.Windows.Forms.DragEventArgs e) {            
			ListBoxControl listBox = sender as ListBoxControl;
			Point newPoint = new Point(e.X, e.Y);
			newPoint = listBox.PointToClient(newPoint);
			int selectedIndex = listBox.IndexFromPoint(newPoint);
			object item = listBoxControl1.Items[listBoxControl1.IndexFromPoint(p)];
			if (selectedIndex == -1)
				listBox.Items.Add(item);
			else
				listBox.Items.Insert(selectedIndex, item);
		}
	}
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data
Imports DevExpress.XtraEditors

Namespace WindowsApplication596
	''' <summary>
	''' Summary description for Form1.
	''' </summary>
	Public Class Form1
		Inherits System.Windows.Forms.Form
		Private WithEvents listBoxControl1 As DevExpress.XtraEditors.ListBoxControl
		Private WithEvents listBoxControl2 As DevExpress.XtraEditors.ListBoxControl
		''' <summary>
		''' Required designer variable.
		''' </summary>
		Private components As System.ComponentModel.Container = Nothing

		Public Sub New()
			'
			' Required for Windows Form Designer support
			'
			InitializeComponent()

			'
			' TODO: Add any constructor code after InitializeComponent call
			'
		End Sub

		''' <summary>
		''' Clean up any resources being used.
		''' </summary>
		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If components IsNot Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		#Region "Windows Form Designer generated code"
		''' <summary>
		''' Required method for Designer support - do not modify
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.listBoxControl1 = New DevExpress.XtraEditors.ListBoxControl()
			Me.listBoxControl2 = New DevExpress.XtraEditors.ListBoxControl()
			CType(Me.listBoxControl1, System.ComponentModel.ISupportInitialize).BeginInit()
			CType(Me.listBoxControl2, System.ComponentModel.ISupportInitialize).BeginInit()
			Me.SuspendLayout()
			' 
			' listBoxControl1
			' 
			Me.listBoxControl1.ItemHeight = 15
			Me.listBoxControl1.Location = New System.Drawing.Point(32, 16)
			Me.listBoxControl1.Name = "listBoxControl1"
			Me.listBoxControl1.Size = New System.Drawing.Size(248, 288)
			Me.listBoxControl1.TabIndex = 0
'			Me.listBoxControl1.MouseMove += New System.Windows.Forms.MouseEventHandler(Me.listBoxControl1_MouseMove);
'			Me.listBoxControl1.MouseDown += New System.Windows.Forms.MouseEventHandler(Me.listBoxControl1_MouseDown);
			' 
			' listBoxControl2
			' 
			Me.listBoxControl2.AllowDrop = True
			Me.listBoxControl2.ItemHeight = 15
			Me.listBoxControl2.Location = New System.Drawing.Point(360, 16)
			Me.listBoxControl2.Name = "listBoxControl2"
			Me.listBoxControl2.Size = New System.Drawing.Size(248, 288)
			Me.listBoxControl2.TabIndex = 0
'			Me.listBoxControl2.DragOver += New System.Windows.Forms.DragEventHandler(Me.listBoxControl2_DragOver);
'			Me.listBoxControl2.DragDrop += New System.Windows.Forms.DragEventHandler(Me.listBoxControl2_DragDrop);
			' 
			' Form1
			' 
			Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
			Me.ClientSize = New System.Drawing.Size(768, 510)
			Me.Controls.Add(Me.listBoxControl1)
			Me.Controls.Add(Me.listBoxControl2)
			Me.Name = "Form1"
			Me.Text = "How to drag/drop ListBoxControl Items between ListBoxControls"
'			Me.Load += New System.EventHandler(Me.Form1_Load);
			CType(Me.listBoxControl1, System.ComponentModel.ISupportInitialize).EndInit()
			CType(Me.listBoxControl2, System.ComponentModel.ISupportInitialize).EndInit()
			Me.ResumeLayout(False)

		End Sub
		#End Region

		''' <summary>
		''' The main entry point for the application.
		''' </summary>
		<STAThread> _
		Shared Sub Main()
			Application.Run(New Form1())
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
			For i As Integer = 0 To 9
				If i Mod 2 = 0 Then
					listBoxControl1.Items.Add("Item " & i.ToString())
				Else
					listBoxControl2.Items.Add("Item " & i.ToString())
				End If
			Next i
		End Sub

		Private p As Point = Point.Empty

		Private Sub listBoxControl1_MouseDown(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles listBoxControl1.MouseDown
			Dim c As ListBoxControl = TryCast(sender, ListBoxControl)
			p = New Point(e.X, e.Y)
			Dim selectedIndex As Integer = c.IndexFromPoint(p)
			If selectedIndex = -1 Then
				p = Point.Empty
			End If
		End Sub

		Private Sub listBoxControl1_MouseMove(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles listBoxControl1.MouseMove
			If e.Button = MouseButtons.Left Then
				If (p <> Point.Empty) AndAlso ((Math.Abs(e.X - p.X) > 5) OrElse (Math.Abs(e.Y - p.Y) > 5)) Then
					listBoxControl1.DoDragDrop(sender, DragDropEffects.Move)
				End If
			End If
		End Sub

		Private Sub listBoxControl2_DragOver(ByVal sender As Object, ByVal e As System.Windows.Forms.DragEventArgs) Handles listBoxControl2.DragOver
			e.Effect = DragDropEffects.Move
		End Sub

		Private Sub listBoxControl2_DragDrop(ByVal sender As Object, ByVal e As System.Windows.Forms.DragEventArgs) Handles listBoxControl2.DragDrop
			Dim listBox As ListBoxControl = TryCast(sender, ListBoxControl)
			Dim newPoint As New Point(e.X, e.Y)
			newPoint = listBox.PointToClient(newPoint)
			Dim selectedIndex As Integer = listBox.IndexFromPoint(newPoint)
			Dim item As Object = listBoxControl1.Items(listBoxControl1.IndexFromPoint(p))
			If selectedIndex = -1 Then
				listBox.Items.Add(item)
			Else
				listBox.Items.Insert(selectedIndex, item)
			End If
		End Sub
	End Class
End Namespace
